BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs

```diff
@@ -134,17 +135,21 @@ private async Task<NewUserResponse> CreateNewUserInAd(string firstname, string l
         private async Task<UserProfile> GetUserByContactEmail(string emailAddress)
         {
             _logger.LogInformation("Attempt to get username by contact email {ContactEmail}.", emailAddress);
+            
+            // Remove diacritics from the email address
+            // We have to do this here rather than in user api as diacritic characters do not get url-encoded by the user api client
+            var sanitisedEmailAddress = emailAddress.ReplaceDiacriticCharacters();
```

Comment:
Is this right? I didn't think we'd be stripping out the characters of their own contact emails, only for the user principle name we create. So when we search by contact email, it should be the same as whatever they provided us?

Reply:
yes unfortunately contact emails can't be stored in AAD with special characters either. Replacing them is the best we can do

Reply:
Sorry - i've double checked aad and realised you're right and it has to be stripped out there aswell.

Reply:
> yes unfortunately contact emails can't be stored in AAD with special characters either. Replacing them is the best we can do.
> 
> we still store their original contact email in the persons db, which is what we use to send hearing notifications. But we have to store a stripped out version of their email in AAD. We strip them out here as well before fetching them from user api, so that the lookup to AAD will work. If that makes sense

Replied before I saw your reply! Sorry, not a problem have approved